app/assets/javascripts/webchat/library.js

```diff
@@ -32,11 +35,139 @@
+              var k2c_launchServe      = 'https://hmpowebchat.klick2contact.com/v03'; // '', 'd' 's'
+
+              //var k2c_staticDept           = "712" // 712 or 717
+              var k2c_staticChnl           = "CH"
+              var k2c_skin                 = "chat_a1"
+              var k2c_staticIID            = "UK"
+              var k2c_lang                 = "en"
+              var k2c_remark               = "K2C"
+              var k2c_StaticCustomString   = ""
+              var k2c_staticFCL            = "0"
+              var k2c_staticQueue          = "0"
+
+              function k2c_getUserCid () {
+                /* Setup Customer ID cookie */
+                var k2c_cid = k2c_getCookie('k2c_'+k2c_provider+"_cids");
+                if (typeof k2c_cid == "undefined") {
+                    k2c_cid = k2c_provider+'_'+k2c_randomString(13);
+                    k2c_setCookie('k2c_'+k2c_provider+"_cids",k2c_cid);
+                }
+                return k2c_cid;
+              }
+
+              function k2c_getCookie(c_name) {
+
+                if (c_name=='k2c_history' && typeof(window.k2c_page)!='undefined') {
+                  return window.k2c_page;
+                }
+
+                var i,x,y,ARRcookies=document.cookie.split(";");
+                for (i=0;i<ARRcookies.length;i++) {
+                  x=ARRcookies[i].substr(0,ARRcookies[i].indexOf("="));
+                  y=ARRcookies[i].substr(ARRcookies[i].indexOf("=")+1);
+                  x=x.replace(/^\s+|\s+$/g,"");
+                  if (x==c_name) {
+                    return unescape(y);
+                  }
+                }
+              }
+
+              function k2c_setCookie(c_name,value,exdays) {
+                if (exdays==null) {
+                  var exdays = 1;
+                }
+                var exdate=new Date();
+                exdate.setDate(exdate.getDate() + exdays);
+                if (exdays!=0) {
+                  var c_value=escape(value) + ((exdays==null) ? "" : "; expires="+exdate.toUTCString()+"; path=/");
+                } else {
+                  var c_value=escape(value) +"; path=/";
+                }
+                document.cookie=c_name + "=" + c_value;
+              }
+
+              function k2c_randomString(length) {
+                var chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+                var chars = chars.split('');
+                var result = '';
+                for (var i = length; i > 0; --i) result += chars[Math.round(Math.random() * (chars.length - 1))];
+                return result;
+              }
+
+               var k2cOpenUrl = k2c_url+'/launcherV3.php?p='+k2c_provider+'&d='+k2c_staticDept+'&ch='+k2c_staticChnl+'&psk='+k2c_skin+'&iid='+k2c_staticIID+'&c='+k2c_getUserCid()+'&l='+k2c_lang+'&u='+encodeURIComponent(window.location.href)+'&r='+k2c_remark+k2c_StaticCustomString+'&fcl='+k2c_staticFCL+'&srbp='+k2c_staticQueue+'&s='+k2c_launchServe;
```

Comment:
This could be tidied up a bit by using a solution like this: https://stackoverflow.com/a/111545/12221146